### PR TITLE
[BugFix] fix setting wrong steamingPreAgg flag

### PR DESCRIPTION
**Pull request by packy92**
Fixes #issue
```
# query 1.
select  /*+ SET_VAR (streaming_preaggregation_mode = 'force_streaming',new_planner_agg_stage='3') */  array_length(array_agg(a1)), count(distinct a1) as cnt from (select split('a,b,c', ',') as a1, 'aaa' as b1) t1 group by b1;

# snippet of plan 
|   7:Project                                           |
|   |  <slot 5> : 5: count                              |
|   |  <slot 6> : array_length(4: array_agg)            |
|   |                                                   |
|   6:AGGREGATE (merge finalize)                        |
|   |  output: array_agg(4: array_agg), count(5: count) |
|   |  group by: 3: expr                                |
|   |                                                   |
|   5:AGGREGATE (update serialize)                      |
|   |  STREAMING                                        |
|   |  output: array_agg(4: array_agg), count(2: split) |
|   |  group by: 3: expr                                |

# query 2
select /*+ SET_VAR (streaming_preaggregation_mode = 'force_streaming',new_planner_agg_stage='3') */ array_length(array_agg(v1)), count(distinct v2) from tbl group by v3;
# snippet of plan 

|   8:Project                                           |
|   |  <slot 5> : 5: count                              |
|   |  <slot 6> : array_length(4: array_agg)            |
|   |                                                   |
|   7:AGGREGATE (merge finalize)                        |
|   |  output: array_agg(4: array_agg), count(5: count) |
|   |  group by: 3: expr                                |
|   |                                                   |
|   6:EXCHANGE                                          |
|                                                       |
| PLAN FRAGMENT 1                                       |
|  OUTPUT EXPRS:                                        |
|   PARTITION: HASH_PARTITIONED: 2: c1, 3: expr         |
|                                                       |
|   STREAM DATA SINK                                    |
|     EXCHANGE ID: 06                                   |
|     HASH_PARTITIONED: 3: expr                         |
|                                                       |
|   5:AGGREGATE (update serialize)                      |
|   |  STREAMING                                        |
|   |  output: array_agg(4: array_agg), count(2: c1)    |
|   |  group by: 3: expr
```
The above two queries cause BE crashed.
The reason is the `useStreamingPreagg` in `AggregationNode` is wrong lead the BE choose streaming operator and the `streaming_preaggregation_mode` hint force using `output_chunk_by_streaming()` when `push_chunk`. However, the `PlanNode 5` in  both queries use a intermediate input from its child  and may fail  to `convert_to_serialize_format()`.

So, we need set `useStreamingPreagg` carefully in FE. Actually, only the first `Local agg` with group by keys and without distinct agg call operator can use streaming mode. The `DISTINCT_LOCAL` agg and `Local` agg above the `DISTINCT_GLOBAL` agg has already implied that the input is low-cardinality, and the choice of preagg is efficient and necessary.

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by satanson — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/implementation/HashAggImplementationRule.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalHashAggregateOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalAggregationOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/tree/AddDecodeNodeForDictStringRule.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/implementation/HashAggImplementationRule.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalHashAggregateOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalAggregationOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/tree/AddDecodeNodeForDictStringRule.java|1|1|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/AggregateTest.java [hunk 1]
```diff
@@ -2328,13 +2322,45 @@ public void testSplitTheTopGlobalAgg() throws Exception {
...
+        assertCContains(plan, "4:AGGREGATE (update serialize)\n" +
+                "  |  STREAMING\n" +
+                "  |  output: sum(4: v4)\n" +
+                "  |  group by: 2: v2, 3: v3",
+                "7:AGGREGATE (update serialize)\n" +
+                        "  |  output: count(2: v2), sum(8: sum)\n" +
+                        "  |  group by: 3: v3");
+
+        sql = "select /*+ SET_VAR (streaming_preaggregation_mode = 'force_streaming') */ " +
+                "count(distinct v2), array_length(array_agg(v1)) from t0 join t1 group by 'a'";
```
Comment (fzhedu) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/AggregateTest.java:2355]: set new_planer_agg_phase = 3 or 4?

**Review by fzhedu — COMMENTED** [review]
are the related tests of query cache passed?

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/AggregateTest.java (hunk 1 above)
Comment (packy92) [reply fe/fe-core/src/test/java/com/starrocks/sql/plan/AggregateTest.java:2355]: These plan are implemented by 3 agg phase.

**Comment by packy92** [comment]
> are the related tests of query cache passed?

Yes, I had checked and add similar tests in the sql test file.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/implementation/HashAggImplementationRule.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalHashAggregateOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalAggregationOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/tree/AddDecodeNodeForDictStringRule.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/implementation/HashAggImplementationRule.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalHashAggregateOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalAggregationOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/tree/AddDecodeNodeForDictStringRule.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/implementation/HashAggImplementationRule.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalHashAggregateOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalAggregationOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/tree/AddDecodeNodeForDictStringRule.java|1|1|100.00%|[]|

**Review by fzhedu — APPROVED** [review]
LGTM

**Review by satanson — APPROVED** [review]

**Pull request merged**